### PR TITLE
Flav/notion api error handling

### PR DESCRIPTION
**Pull request by flvndvd**
This PR resolves https://github.com/dust-tt/dust/pull/3146#pullrequestreview-1815281941 by handling properly the "unauthorized" error when searching the top-level pages.

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 1]
```diff
@@ -229,6 +238,7 @@ export async function getPagesAndDatabasesToSync({
 
   const localLogger = logger.child({
     ...loggerArgs,
+    connectorId: connector.id,
```
Comment (flvndvd) [inline connectors/src/connectors/notion/temporal/activities.ts:241]: 🚗

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 2]
```diff
@@ -261,31 +268,33 @@ export async function getPagesAndDatabasesToSync({
...
-            attempt: Context.current().info.attempt,
-          },
-          "Failed to get Notion search result page with cursor. Giving up and moving on"
-        );
-        return {
-          pageIds: [],
-          databaseIds: [],
-          nextCursor: null,
-        };
+    if (e instanceof Error && isNotionError(e)) {
```
Comment (flvndvd) [inline connectors/src/connectors/notion/temporal/activities.ts:271]: Made sure that NotionHQ.Client() uses `Error` in their base errors.

https://github.com/makenotion/notion-sdk-js/blob/ede416af263e8a866a19b912bcd022369e8009a9/src/errors.ts#L38-L42

File: connectors/src/connectors/notion/lib/errors.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// This type is a simplified abstraction and
+// does not fully represent the structure of errors returned by the Notion API.
+interface NotionError extends Error {
+  body: unknown;
+  code: string;
+  status: number;
+}
+
+export function isNotionError(error: Error): error is NotionError {
+  return "code" in error;
```
Comment (spolu) [inline connectors/src/connectors/notion/lib/errors.ts:10]: Is that enough? that could be true of many other errors no?

File: connectors/src/connectors/notion/temporal/activities.ts (hunk 2 above)
Comment (spolu) [reply connectors/src/connectors/notion/temporal/activities.ts:267]: Do they all have `code` ? is `code` really a good probe for this?

Maybe we should test that code is either "internal_server_error" | "unauthorized" in isNotionError?

**Review by spolu — APPROVED** [review]

**Pull request merged**